pyk/src/pyk/proof/reachability.py

```diff
@@ -44,7 +44,19 @@ class APRProofResult:
 
 @dataclass
 class APRProofExtendResult(APRProofResult):
-    extend_result: KCFGExtendResult
+    """Holds the description of how an APRProof should be extended.
+
+    Fields:
+        extend_results: Holds the KCFG extension to be applied.
+        to_cache: Holds an indicator of whether or not the provided extension
+                  should be cached instead of being applied.
```

Comment:
`extend_cterm` can now return two extensions, the first of which is a `Step`, and the second of which is a cut-rule, branch, or NDBranch. When that happens, the first one is to be applied, and the second one is to be cached so that it can be automatically applied instead of recomputed when we explore from the node created by that first `Step`.